readthedocs/projects/models.py

```diff
@@ -570,8 +572,15 @@ def has_aliases(self):
         return self.aliases.exists()
 
     def has_pdf(self, version_slug=LATEST):
+        version = self.versions.filter(slug=version_slug).first()
         if not self.enable_pdf_build:
```

Comment:
Yea, the tests would need to be changed, as it would be a change in how things work. Perhaps the easiest update would be to actually update the database for a project from the YAML file on build. I'd want @agjohnson's thoughts on that though, because it is a weird UX. Perhaps if we detect YAML we can hide the fields on the web UI, and update them from the YAML for consistency.